### PR TITLE
#26: Update front-end section (closes #26)

### LINKED ISSUE #26: Update front-end section
## requirements
Front-end section should be put up to date with the current state of the tools, libraries and guidelines, and the *tutorial* section should be updated to match the `tutorial` branch in `webseed`.

## specs
{optional: describe technical specs to implement this feature, if not obvious}

## misc
{optional: other useful info}

### PR DESCRIPTION
**Pull request by ecamellini**
Closes #26

* Fixed some typos around
* Updated front-end section and added mention to scriptoni
* Updated tutorial

### CODE REVIEW

File: book/collaboration/1.slack.md [hunk 1]
```diff
@@ -17,15 +17,15 @@ Team awareness is extremely important, so - by default - prefer writing in a pub
 Even if you are talking to a specific person the information could be useful to other people, even when you don't realize it.
 You can still direct your question to a specific person using [**mentions**](https://get.slack.help/hc/en-us/articles/205240127-Mention-a-team-member).
 
-Remember that you can **create new public channels** easily; if you are afraid to "spam" people, you can create a channel for the topic you want to discuss and move the conversation there. Channels can then be archived when their purpose is over (e.g. a channel about an event, that is now passed).
+Remember that you can **create new public channels** easily; if you are afraid to "spam" people, you can create a channel for the topic you want to discuss and move the conversation there. Channels can then be archived when their purpose is over (e.g.. a channel about an event, that is now passed).
```
Comment (giogonzo) [inline book/collaboration/1.slack.md]: ?

File: book/frontend/5.tutorial.md [hunk 2]
```diff
@@ -1,23 +1,44 @@
...
 
 ### Step 2
 
-Using Containers and `@skinnable` to separate data and rendering
+Using Containers to separate data and rendering.
 
 ### Step 3
 
-Fetching data from an API using Avenger
+Using to separate logic and UI.
```
Comment (giogonzo) [inline book/frontend/5.tutorial.md]: broken phrase?

File: book/frontend/guidelines/1.general_structure.md [hunk 3]
```diff
@@ -32,7 +32,7 @@ All app source code should stay in `<project root>/src`:
 
 `src/app` is where the core app code goes. This should be the majority of source code.
 
-`src/client` contains client-specific bootstrap files, typically 1:1 with a [webpack `entry` config](https://webpack.github.io/docs/configuration.html#entry). We use a folder here since you might have multiple build configurations, to produce different apps sharing source code, or to produce bundles for different targets (e.g. website vs electron).
+`src/client` contains client-specific bootstrap files, typically 1:1 with a [webpack `entry` config](https://webpack.github.io/docs/configuration.html#entry). We use a folder here since you might have multiple build configurations, to produce different apps sharing source code, or to produce bundles for different targets (, website vs electron).
```
Comment (giogonzo) [inline book/frontend/guidelines/1.general_structure.md]: ?

File: book/frontend/guidelines/4.CSS.md [hunk 4]
```diff
@@ -135,11 +137,11 @@ The `/theme` folder should contain "global" SASS files and a `variables.scss` wi
 
 What should be part of variables.scss?
 
-Rule of thumb for now has been "*everything with a name on Zeplin*":
+Rule of thumb for now has been "*everything with a name in the design*" (usually on [Zeplin](https://zeplin.io/), where some projects also have a specific `styleguide` with everything that should be put in `variables.scc`, `typography.scss` etc.):
```
Comment (giogonzo) [inline book/frontend/guidelines/4.CSS.md]: .scss

File: book/frontend/tutorial/step_3.md [hunk 5]
```diff
@@ -66,5 +69,8 @@ const { greeting } = instance.getLocals(props);
 expect(greeting).toBe("Hello");
 ```
 
-## Step2->Step3
-Check the [full diff](https://github.com/buildo/webseed/compare/tutorial-step2...tutorial-step3) between Step 2 and Step 3.
+Despite this, it is not something we do it often in practite. But it's still good to know.
```
Comment (giogonzo) [inline book/frontend/tutorial/step_3.md]: `we do it`

File: book/frontend/tutorial/step_4.md [hunk 6]
```diff
@@ -4,7 +4,7 @@
 
 Most web applications need to frequently synchronize data with a server using HTTP calls.
 
-To manage these calls in a simple and consistent manner, we have created [Avenger](https://github.com/buildo/avenger). It's a powerful data-fetching library supporting batching, caching, data dependencies and much more. Refer to the [Avenger book](https://buildo.gitbooks.io/avenger/content/) for more details.
+To manage these calls in a simple and consistent manner, we created [Avenger](https://github.com/buildo/avenger). It's a powerful data-fetching library supporting batching, caching, data dependencies and much more. Refer to the [Avenger book](https://buildo.gitbooks.io/avenger/content/) for more details.
```
Comment (giogonzo) [inline book/frontend/tutorial/step_4.md]: I'd avoid linking to the book which is very outdated

File: book/frontend/tutorial/step_6.md [hunk 7]
```diff
@@ -1,43 +1,55 @@
...
 ```js
-const myLoadingComponent = loading({
-  wrapper: <div style={{ textAlign: 'center', position: 'relative', minHeight: 100 }} />,
-  loader: <div>loading...</div>
-});
+export default from './LoadingSpinner';
 ```
 
-## Using the LoadingSpinner UI component
+3. Import it's style in the `loadingSpinner.scss` file:
```
Comment (giogonzo) [inline book/frontend/tutorial/step_6.md]: `its`

File: book/frontend/tutorial/step_7.md [hunk 8]
```diff
@@ -1,42 +1,104 @@
...
+.panel {
+  border-radius: 10px;
+  box-shadow: 0 2px 0 $darkGrey;
+
+  .panel-content {
+    border-radius: 10px;
+  }
+}
 ```
+We just overridden its default background color and customized its border.
```
Comment (giogonzo) [inline book/frontend/tutorial/step_7.md]: I think this override can be done via variables defined in original panel scss? If so, it'd be better to show a customization via sass variables override

File: book/frontend/tutorial/step_8.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
+# Step 8
+
+You reached the last step of the tutorial!
+
+In our projects we almost always use the (FlexView)[https://github.com/buildo/react-flexview] component to compose our layout and to wrap our components, instead of the classic `<div>`. Let's try to do the same here!
```
Comment (giogonzo) [inline book/frontend/tutorial/step_8.md]: adding rationale behind this choice would be great :)

**Review by giogonzo — CHANGES_REQUESTED** [review]
some typos and some other minor comments inline, overall 👍

File: book/collaboration/1.slack.md (hunk 1 above)
Comment (ecamellini) [reply book/collaboration/1.slack.md]: Typone, should be a comma

File: book/frontend/5.tutorial.md (hunk 2 above)
Comment (ecamellini) [reply book/frontend/5.tutorial.md]: Broken, fixing this

File: book/frontend/guidelines/1.general_structure.md (hunk 3 above)
Comment (ecamellini) [reply book/frontend/guidelines/1.general_structure.md]: Typone

File: book/frontend/tutorial/step_7.md (hunk 8 above)
Comment (ecamellini) [reply book/frontend/tutorial/step_7.md]: I do it for the `$content-background: $cloud;`!
 
Shadow and border-radius are not defined in the original panel.scss I think, so for these I just used normal css style

Comment (giogonzo) [reply book/frontend/tutorial/step_7.md]: :+1:

File: book/frontend/tutorial/step_8.md (hunk 9 above)
Comment (ecamellini) [reply book/frontend/tutorial/step_8.md]: I can link to the intro of the book of flexview, that shows an example of why it is better!

Comment (giogonzo) [reply book/frontend/tutorial/step_8.md]: :+1:

**Comment by ecamellini** [comment]
@giogonzo addressed all comments and fixed some more typos.

**Review by giogonzo — APPROVED** [review]

**Pull request merged**